pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -291,9 +291,11 @@ def pre_configure_ddp(self):
             self._ddp_kwargs["find_unused_parameters"] = True
 
     def _register_ddp_hooks(self) -> None:
-        # currently, DDP communication hooks only work with NCCL backend and SPSD (single process single device) mode
-        # https://github.com/pytorch/pytorch/blob/v1.8.0/torch/nn/parallel/distributed.py#L1080-L1084
-        if _TORCH_GREATER_EQUAL_1_8 and self.on_gpu and self._is_single_process_single_device:
+        # In 1.8, DDP communication hooks only work with NCCL backend and SPSD (single process single device) mode
+        # Since 1.9, DDP communication hooks can work on all backends.
+        if _TORCH_GREATER_EQUAL_1_9 or (
```

Comment:
this needs to be imported
did you run the code? 😅

Reply:
Thanks for the review and question!

> this needs to be imported
> did you run the code? 😅

Good catch! Nope, I don't know I need to import it. This is my first Lighting PR. I work on PyTorch DDP, but I don't know how to run code...

Reply:
That's awesome, thanks for the PR